src/qt/ve_qitems_mqtt.cpp

```diff
@@ -183,14 +183,21 @@ void VeQItemMqttProducer::open(
 		mWebSocket->setProtocol(
 			  protocolVersion == QMqttClient::MQTT_3_1 ? "mqttv3.1" : "mqtt");
 
+		connect(mWebSocket, &WebSocketDevice::disconnected,
+			this, [this] {
+				onStateChanged(QMqttClient::Disconnected);
+			}, Qt::QueuedConnection);
+
 		connect(mWebSocket, &WebSocketDevice::connected,
 			this, [this, protocolVersion] {
 				mMqttConnection->setProtocolVersion(protocolVersion);
 				mMqttConnection->setTransport(mWebSocket, QMqttClient::IODevice);
 				QMetaObject::invokeMethod(this, [this] { Q_EMIT aboutToConnect(); }, Qt::QueuedConnection);
-			});
+			}, Qt::QueuedConnection);
```

Comment:
Any particular reason for the Qt::QueuedConnection? The sender and the receiver are both in the same thread, no? Here and elsewhere